### ISSUE TITLE
How to add artwork? / Extension still working?

### ISSUE THREAD

**Issue by jvolker** [karlosgliberal/Openframe-Processing#1]
I've tried adding this Processing example https://raw.githubusercontent.com/shiffman/LearningProcessing/master/chp01_drawing/example_01_03_rgbcolor/example_01_03_rgbcolor.pde by entering the URL in a new artwork on the web app of Openframe. 

Unfortunately, I am left with a black screen when trying to display it on the screen.

How are artworks meant to be added?
Is the extension still working?

Thanks.

**Comment by jvolker** [comment]
I've fixed the extension: https://groups.google.com/forum/#!topic/openframeio/xBd-9ZUGsz4
I'm just doing some more testing and will get back with a pull request.

**Comment by jeremydouglass** [comment]
@jvolker -- currently the only instructions on installation are `openframe -i openframe-processing`. How should we install your fork?

For example, I tried installing it locally through nvm install --save jvolker/Openframe-Processing, but I'm not sure if this had any effect on openframe -- I'm still getting a java.awt.HeadlessException: "No X11 DISPLAY variable was set"

...I see that there are issues open on the main OpenframeProject/Openframe repo about this -- and it looks like the current options are either creating a full dev setup or using a published, non-scoped npm package...?

**Comment by jvolker** [comment]
Yeah, you are right. I believe the only way at the moment is to do a dev setup as described in 
http://docs.openframe.io/dev-env-setup/.  I think it's not possible to install a scoped or unpublished package currently, as described in https://github.com/OpenframeProject/Openframe/issues/49. I've started [a fork](https://github.com/jvolker/Openframe/tree/install-extensions-from-github) to support this but it's sort of untested so far. I got errors and didn't know if this had to do with my dev setup or my changes in the code. I'm going to get back to https://github.com/OpenframeProject/Openframe/issues/56 soon. Maybe once that is solved we can work on installing from Github and scoped packages as well. Would be good to have this option for testing purposes anyway.

That said, I contacted the owner of this extension package @karlosgliberal and asked him about the future of this extension and if he would like to stay involved in it. I haven't heard back yet. Originally, I intended to do a pull request (hence the install instructions) but I'm also happy to continue as the main maintainer and taking over the NPM package. I would like to avoid publishing to a new NPM package as this would leave behind an unusable package with a recognizable name as well as a working package with a rather cumbersome name. New users would probably be confused. Again, happy to do a pull request only.

**Comment by jeremydouglass** [comment]
Thanks. Yes -- a full dev setup is a bit much for us to set up and maintain just to load a single dev extension, and the ability to load non-standard extensions is critical if people are going to carry on extension development in forks (as they often do). I'll comment on OpenframeProject/Openframe#49
